ihmc-robotics-toolkit/src/main/java/us/ihmc/robotics/screwTheory/FloatingBaseRigidBodyDynamicsCalculator.java

```diff
@@ -4,133 +4,281 @@
 import org.ejml.dense.row.CommonOps_DDRM;
 import org.ejml.dense.row.factory.LinearSolverFactory_DDRM;
 import org.ejml.interfaces.linsol.LinearSolverDense;
+import us.ihmc.commons.MathTools;
+import us.ihmc.robotics.MatrixMissingTools;
 
 public class FloatingBaseRigidBodyDynamicsCalculator
 {
    private static final int large = 1000;
    private static final double tolerance = 0.0001;
 
-   private static final LinearSolverDense<DMatrixRMaj> pseudoInverseSolver = LinearSolverFactory_DDRM.pseudoInverse(true);
+   private final LinearSolverDense<DMatrixRMaj> pseudoInverseSolver = LinearSolverFactory_DDRM.pseudoInverse(true);
 
    private final DMatrixRMaj matrixTranspose = new DMatrixRMaj(large, large);
-   private final DMatrixRMaj localVector = new DMatrixRMaj(large, 1);
+   private final DMatrixRMaj tempObjective = new DMatrixRMaj(large, large);
 
    public FloatingBaseRigidBodyDynamicsCalculator()
```

Comment:
This class was also a travesty and could have been used as a "how not to write code" example.